.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,70 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
+  schedule:
+    - cron: '37 15 * * 5'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'python' ]
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby' ]
+        # Learn more about CodeQL language support at https://git.io/codeql-language-support
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
```

Comment:
Given the [warning](https://github.com/Sage-Bionetworks-Workflows/cwl-tool-template/runs/5206023994?check_suite_focus=true#step:3:269) that's raised by CodeQL (see screenshot below), we might need to pin the version of Python to match what's in the Pipfile (or update the Pipfile to use whatever version is included with `ubuntu-latest`, which will presumably change over time and thus less ideal). 

```suggestion

    - uses: actions/setup-python@v2
      with:
        python-version: '3.7'

```

![image](https://user-images.githubusercontent.com/740725/154176556-eb583ad4-0603-4c40-ba24-006119d35ff3.png)

Reply:
Oddly, I don't see that warning in the logs you linked, but I am accepting the suggestion.